network/yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -181,4 +249,44 @@ public ByteBuffer getMetaData() {
     return ByteBuffer.allocate(0);
   }
 
+  private Map<String, List<Entry<AppExecId, ExecutorShuffleInfo>>> reloadRegisteredExecutors()
+      throws IOException, ClassNotFoundException {
+    if (registeredExecutorFile != null && registeredExecutorFile.exists()) {
+      logger.info("Reloading executors from {}", registeredExecutorFile);
+      return reloadRegisteredExecutors(registeredExecutorFile);
+    } else {
+      logger.info("No executor info to reload");
+      return new HashMap<>();
+    }
+  }
+
+
+  @VisibleForTesting
+  static Map<String, List<Entry<AppExecId, ExecutorShuffleInfo>>> reloadRegisteredExecutors(
+      File file
+  ) throws IOException, ClassNotFoundException {
+    Map<String, List<Entry<AppExecId, ExecutorShuffleInfo>>> result = new HashMap<>();
+    ObjectInputStream in = null;
+    try {
+      in = new ObjectInputStream(new FileInputStream(file));
+      Map<AppExecId, ExecutorShuffleInfo> registeredExecutors =
```

Comment:
Sean was cleaning up `-Xlint` warning, so it would be nice to silence this one (`@SuppressWarnings("unchecked")`).